From 3decaea196648b60b923a649c3c82d3f2a12e4e5 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sat, 25 Nov 2023 00:23:22 +0100
Subject: [PATCH 22/26] clean up after depmod if modules.dep is empty

we just removed the last module from (no longer installed) kernel $1
so do not leave stale depmod files around
---
 dkms.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dkms.in b/dkms.in
index a7d46f1..095178a 100644
--- a/dkms.in
+++ b/dkms.in
@@ -334,6 +334,13 @@ do_depmod()
     else
         depmod -a "$1"
     fi
+    if [ -f /lib/modules/$1/modules.dep ] && [ ! -s /lib/modules/$1/modules.dep ]; then
+        # if modules.dep is empty, we just removed the last kernel module from
+        # no longer installed kernel $1, so do not leave stale depmod files around
+        rm -fv /lib/modules/$1/modules.{alias,dep,devname,softdep,symbols,*.bin}
+        rmdir --ignore-fail-on-non-empty /lib/modules/$1
+        test -d /lib/modules/$1 || echo $"Removed /lib/modules/$1"
+    fi
 }
 
 # Grab distro information from os-release.
-- 
2.20.1

